### PR TITLE
Subview docs for programming guide

### PR DESCRIPTION
**Pull request by nmm0**
Add documentation for the `Subview` type alias in the programming guide

This also adds math notation extension to the markdown `myst` parser

### CODE REVIEW

File: source/ProgrammingGuide/Subviews.md [hunk 1]
```diff
@@ -71,3 +71,14 @@ Suppose that a View has `k` dimensions. Then when calling `subview` on that View
 ### 11.2.3 Degenerate Views
 
 Given a View with `k` dimensions, we call that View _degenerate_ if any of its dimensions is zero. Degenerate Views are useful for keeping code simple by avoiding special cases. For example, consider a MPI (Message-Passing Interface) distributed-memory parallel code that uses Kokkos to represent local (per-process) data structures. Suppose that the code distributes a dense matrix (2-D array) in block row fashion over the MPI processes in a communicator. It could be that some processes own zero rows of the matrix. This may not be efficient, since those processes do no work yet participate in collectives, but it might be possible. In this case, allowing Views with zero rows would reduce the number of special cases in the code.
+
+## 11.3 Obtaining the type of a subview
+
+For some applications, it is useful to know the type of a subview, given the arguments. In these cases, using `auto` type deduction can be inconvenient. While it is possible to use `decltype()` and `std::declval` to obtain this from the `subview` function, Kokkos provides a convenient type alias:
+
+```c++
+using my_view_type = View<double **>;
```
Comment (JBludau) [inline source/ProgrammingGuide/Subviews.md:80]: This should be `Kokkos::View<double **>`;

**Review by JBludau — APPROVED** [review]
looks good

**Comment by fnrizzi** [comment]
@nmm0 sorry for trouble, but can you please rebase and post PR against main? (the location of source has changed in main)

**Comment by nmm0** [comment]
Rebased and should be ready to merge

File: docs/source/ProgrammingGuide/Subviews.md [hunk 2]
```diff
@@ -10,20 +10,20 @@ After reading this chapter, you should understand the following:
...
+structured subset of the original array. "Behaves as an array" means that the slice has the same syntax as an array does; one can access its entries using array indexing notation. "View" means that the slice and the original array point to the same data, i.e, the slice sees changes to the original array and vice versa. "Structured subset" means a cross product of indices along each dimension, as for example a plane or face of a cube. If the original array has dimensions $\left(N_0, N_1, ..., N_{k-1}\right)$, then a slice views all entries whose indices are $\left(a_0, a_1, ..., a_{k-1}\right)$, where $a_j$ is an ordered subset of $\left(N_0, N_1, ..., N_j-1\right)$.
```
Comment (dalg24) [inline docs/source/ProgrammingGuide/Subviews.md:13]: Please confirm you meant to change the curly braces to regular parentheses in the last change on this line

File: docs/source/ProgrammingGuide/Subviews.md [hunk 3]
```diff
@@ -71,3 +71,14 @@ Suppose that a View has `k` dimensions. Then when calling `subview` on that View
...
+
+## 11.3 Obtaining the type of a subview
+
+For some applications, it is useful to know the type of a subview, given the arguments. In these cases, using `auto` type deduction can be inconvenient. While it is possible to use `decltype()` and `std::declval` to obtain this from the `subview` function, Kokkos provides a convenient type alias:
+
+```c++
+using my_view_type = Kokkos::View<double **>;
+using my_subview_type = Kokkos::Subview<my_view_type,
+                                        Kokkos::ALL,
+                                        Kokkos::pair<unsigned, unsigned>>;
```
Comment (dalg24) [inline docs/source/ProgrammingGuide/Subviews.md:83]: using my_subview_type_deduced = decltype(subview(std::declval<my_view_type>(), Kokkos::ALL, std::make_pair(1u, 1u));
static_assert(std::is_same<my_subview_type, my_subview_type_deduced>::value);

File: docs/source/ProgrammingGuide/Subviews.md (hunk 2 above)
Comment (nmm0) [reply docs/source/ProgrammingGuide/Subviews.md:13]: no, that was a typo, thanks for catching

File: docs/source/ProgrammingGuide/Subviews.md (hunk 3 above)
Comment (nmm0) [reply docs/source/ProgrammingGuide/Subviews.md:83]: will add

**Comment by JBludau** [comment]
@dalg24 @crtrott please reconsider this for merging as the discussion in [#5386](https://github.com/kokkos/kokkos/pull/5386) would favor keeping it in which case it should have documentation.

**Comment by fnrizzi** [comment]
@JBludau @nmm0 is this updated and ready?

File: docs/source/ProgrammingGuide/Subviews.md [hunk 4]
```diff
@@ -71,3 +71,18 @@ Suppose that a View has `k` dimensions. Then when calling `subview` on that View
...
+
+## 11.3 Obtaining the type of a subview
+
+For some applications, it is useful to know the type of a subview, given the arguments. In these cases, using `auto` type deduction can be inconvenient. While it is possible to use `decltype()` and `std::declval` to obtain this from the `subview` function, Kokkos provides a convenient type alias:
+
+```c++
+using my_view_type = Kokkos::View<double **>;
+using my_subview_type = Kokkos::Subview<my_view_type,
+                                        Kokkos::ALL,
```
Comment (dalg24) [inline docs/source/ProgrammingGuide/Subviews.md:82]:
```suggestion
                                        decltype(Kokkos::ALL),
```

Comment (JBludau) [reply docs/source/ProgrammingGuide/Subviews.md:82]: yeah, we definitely forgot this

**Comment by fnrizzi** [comment]
@nmm0 can you please finalize this when you can?

**Pull request merged**